data/tutorials/getting-started/02_fst_program.md

```diff
@@ -0,0 +1,308 @@
+
+## Compiling OCaml Programs
+
+OCaml comes with both an interpreter and a compiler. In the previous tutorial, only the interpreter was used. This section gives a glimpse of how to use the compiler. 
+
+We start by seting up a traditional “Hello World!” project using Dune, OCaml’s build system. The following creates it under the name `hello`:
+
+```shell
+$ dune init proj hello
+Entering directory '/home/cuihtlauac/caml/ocaml.org/hello'
+Success: initialized project component named hello
+```
+
+The project is stored in a directory named `hello`. The `tree` command lists the files and directories created. 
+
+```shell
+$ cd hello
+$ tree
+.
+├── bin
+│   ├── dune
+│   └── main.ml
+├── _build
+│   └── log
+├── dune-project
+├── hello.opam
+├── lib
+│   └── dune
+└── test
+    ├── dune
+    └── hello.ml
+
+4 directories, 8 files
+```
+
+OCaml source files have `.ml` extension, which stands for “Meta Language.” Meta Language (ML) is the ancestor of OCaml, this is also what the “ml” stands for in “OCaml.” Here is the content of the `bin/main.ml` file:
+
+```shell
+$ cat ./bin/main.ml
+let () = print_endline "Hello, World!"
+```
+
+The project-wide metadata is available in the `dune-project` file. Each folder containing source files that need to be built must contain a `dune` file explaining how.
+
+This builds the project:
+```shell
+$ dune build
+Entering directory '/home/cuihtlauac/caml/ocaml.org'
+```
+
+This launches the executable it defines:
+```shell
+$ dune exec hello
+Entering directory '/home/cuihtlauac/caml/ocaml.org'
+Hello, World!
+```
+
+Edit the `bin/main.ml` file. Replace `World` by your first name. Recompile the project. Launch it againg. Here your are, this is your first OCaml program.
+
+In the rest of this tutorial, we will make changes to this project to illustrate OCaml tooling and 
+
+## Why no Main Function? 
+
+Although `bin/main.ml`'s name accurately suggests it contains the application entry point into the executable binary produced by the project, it does not contain a dedicated `main` function. Even that file name is irrelevant, it could be anything. This is because OCaml is equally an interpreted and compiled language. Once compiled, binaries are executed as if their source counterpart where read.
+
+That's why double semicolumns are no longer needed in source files, statements are just processed in order, from top to bottom, each triggering the side effects it may have. Definitons are added to the environment. Values resulting from nameless expressions are ignored. Side effects from all those will take place in the same order. That's OCaml main.
+
+However, it is common practice to single out a value that triggers all the side effects and mark it as the intended main entry point. In OCaml, that's the role of the `let () =` pattern, which stands for “definition without a name” and actually indicates we're only interested in the side effects of the value on the right.
```

Comment:
What exactly stands for "definition without a name"? The `()` or the entire expression `let () =`?